Shared/Samples/Set up location-driven geotriggers/SetUpLocationDrivenGeotriggersView.swift

```diff
@@ -0,0 +1,336 @@
+    
+    var body: some View {
+        MapView(map: model.map)
+            .locationDisplay(model.locationDisplay)
+            .task {
+                // Start geotrigger monitoring once the map loads.
+                do {
+                    try await model.map.load()
+                    model.startGeotriggerMonitoring()
+                } catch {
+                    model.error = error
+                }
+            }
+            .overlay(alignment: .top) {
+                // Status text overlay.
+                VStack {
+                    Text("\(model.fenceGeotriggerText)\n")
+                        .frame(maxWidth: .infinity, alignment: .leading)
+                    
+                    Text(model.nearbyFeaturesText)
+                        .foregroundColor(.orange)
+                        .frame(maxWidth: .infinity, alignment: .leading)
+                        .padding(.top, -24)
+                }
+                .padding(8)
+                .background(.thinMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .toolbar {
+                // Bottom button toolbar.
+                ToolbarItemGroup(placement: .bottomBar) {
+                    Button("Current Section") {
+                        let sectionFeature = model.nearbyFeatures[model.currentSectionName!]!
+                        popup = Popup(geoElement: sectionFeature)
+                        isShowingPopup = true
+                    }
+                    .disabled(!model.hasCurrentSection)
+                    .opacity(isShowingPopup ? 0 : 1)
+                    
+                    Button("Point of Interest") {
+                        let poiFeature = model.nearbyFeatures[model.nearbyPOINames.first!]!
+                        popup = Popup(geoElement: poiFeature)
+                        isShowingPopup = true
+                    }
+                    .disabled(!model.hasPointOfInterest)
+                    .opacity(isShowingPopup ? 0 : 1)
+                }
+            }
+            .floatingPanel(
+                selectedDetent: .constant(.full),
+                horizontalAlignment: .leading,
+                isPresented: $isShowingPopup
+            ) {
+                Group {
+                    // Feature info popup.
+                    if let popup = popup {
+                        PopupView(popup: popup, isPresented: $isShowingPopup)
+                            .showCloseButton(true)
+                    }
+                }
+                .padding()
+            }
+            .task(id: isShowingPopup) {
+                // Stop location updates when the popup is showing.
+                if isShowingPopup {
+                    await model.locationDisplay.dataSource.stop()
+                } else {
+                    do {
+                        try await model.locationDisplay.dataSource.start()
+                    } catch {
+                        model.error = error
+                    }
```

Comment:
Very unlikely to throw an error as the location data source is just a string, so I would use `try?` to omit the error here.